packages/material-ui-lab/src/SpeedDialAction/SpeedDialAction.js

```diff
@@ -37,13 +37,26 @@ class SpeedDialAction extends React.Component {
   };
 
   handleTooltipClose = () => {
+    if (this.props.tooltipOpen) return;
     this.setState({ tooltipOpen: false });
   };
 
   handleTooltipOpen = () => {
+    if (this.props.tooltipOpen) return;
     this.setState({ tooltipOpen: true });
```

Comment:
@hashwin I was just looking at this and this seems brittle since other methods can still change the state of tooltipOpen. Not sure if we can do anything about this though.

Reply:
I see your point, but yeah I don't really see another way of handling this without the state and state itself can be changed from anywhere in the component. In fact, we need to manually set the state here for this feature to work: https://github.com/mui-org/material-ui/pull/12590/files#diff-3c8c197bf9114d5f5a42f05daae04497R59

One thing we can maybe do here is to change the method name from ```handleTooltipOpen``` to something like ```onHover``` so that it explicitly says that nothing should be done on hover if the prop ```tooltipOpen``` is given. Thoughts?